src/actors/common/cache.rs

```diff
@@ -2,19 +2,28 @@ use std::collections::BTreeMap;
 use std::fmt;
 use std::io;
 use std::path::{Path, PathBuf};
+use std::pin::Pin;
 use std::sync::Arc;
 
-use actix::ResponseFuture;
-use futures01::future::{self, Future, Shared};
-use futures01::sync::oneshot;
+use futures::channel::oneshot;
+use futures::future::{self, Future, FutureExt, Shared, TryFutureExt};
 use parking_lot::Mutex;
+use sentry::{Hub, SentryFutureExt};
 use symbolic::common::ByteView;
 use tempfile::NamedTempFile;
 
 use crate::cache::{get_scope_path, Cache, CacheKey, CacheStatus};
 use crate::types::Scope;
 use crate::utils::futures::CallOnDrop;
-use crate::utils::sentry::SentryFutureExt;
+
+/// A pinned, boxed future.
+///
+/// This is the type of future that [`futures::FutureExt::boxed`] would return.  This is
+/// pretty boring but clippy quickly complains about type complexity without this.
+pub type BoxedFuture<T> = Pin<Box<dyn Future<Output = T>>>;
```

Comment:
This can probably be declared in some utility module since it's a pattern we'll use more in this code base.